ddl/column.go

```diff
@@ -1102,6 +1102,138 @@ func (w *worker) doModifyColumnTypeWithData(
 	return ver, errors.Trace(err)
 }
 
+func checkOffsets(tbl *model.TableInfo) {
+	for _, col := range tbl.Columns {
+		for _, idx := range tbl.Indices {
+			for _, idxCol := range idx.Columns {
+				if col.Name.L != idxCol.Name.L {
+					continue
+				}
+				if col.Offset != idxCol.Offset {
+					logutil.BgLogger().Error("table meta data is corrupted",
+						zap.String("column name", col.Name.O),
+						zap.Int("table column offset", col.Offset),
+						zap.Int("index column offset", idxCol.Offset))
+				}
+			}
+		}
+	}
+}
+
+func replaceOldColumn(tblInfo *model.TableInfo, oldID, newID int64,
+	newName model.CIStr) (changingCol *model.ColumnInfo, err error) {
+	// Locate the old column and new column.
+	oldCol := model.FindColumnInfoByID(tblInfo.Columns, oldID)
+	if oldCol == nil {
+		return nil, infoschema.ErrColumnNotExists.GenWithStackByArgs(oldCol.Name, tblInfo.Name)
+	}
+	changingCol = model.FindColumnInfoByID(tblInfo.Columns, newID)
+	if changingCol == nil {
+		return nil, infoschema.ErrColumnNotExists.GenWithStackByArgs(oldCol.Name, tblInfo.Name)
+	}
+	// Replace the old column.
+	tblInfo.Columns[oldCol.Offset] = changingCol
+	tblInfo.Columns[changingCol.Offset] = nil
+	changingCol = updateChangingCol(changingCol, newName, oldCol.Offset)
+	// Remove nil column.
+	tmp := tblInfo.Columns[:0]
+	for _, c := range tblInfo.Columns {
+		if c != nil {
+			tmp = append(tmp, c)
+		}
+	}
+	tblInfo.Columns = tmp
+	return changingCol, nil
+}
+
+func replaceOldIndexes(tblInfo *model.TableInfo, changingIdxs []*model.IndexInfo) {
```

Comment:
move this func to `index.go`

Reply:
If it is only used by `modify column`, I think this should be fine?